Jenkinsfile.nightly

```diff
@@ -110,6 +111,17 @@ pipeline {
                 }
             }
         }
+        stage('Nightly-Docker-Image') {
+            steps {
+                withDockerRegistry([ credentialsId: "NIGHTLY_DOCKER_HUB_CRED", url: "" ]) {
+                    sh 'make docker-image'
```

Comment:
hmm, don't we already have the image built somewhere as part of the tests? It's more obvious to build it here, but it might be simpler to tag the existing one and push

Reply:
oh, I guess this was to @eloycoto since he suggested it

Reply:
@raybejjani unfortunately it's built inside vms which after tests are run are deleted via gingkgo framework.

First version of this PR was meant to copy the image from vm to host, but as @eloycoto pointed out before - the vms are already deleted after tests are run.

I personally believe that we should build the image on host and send it to vms so it's built only once, but this will need some bigger changes in the way our CI works (currently it's part of `vagrant provision`).

Reply:
boo :( I guess it will build again then. shrug

Reply:
I mean, if you consider that can be build in the VM and get that, I'm happy with that, but: 

- We lost the stage part, and the Jenkins variables in there. We need to make changes in the Vagrantfiles. 
- The password will send to Vagrantbox, and we can mess-up, not now, but in the future. 
- With this one, it is easy to export only when happens on master. 

The thing is, if the nightly is taking 6 hours, 6 hours and 5 min will be ok from my point of view. Something that can been improved, and it'll help a lot is to not clean the build before do `make docker-image` that will help a lot in normal CI and will make things faster.

Reply:
Yeah. It was more that we upload the same image we tested, in case we needed to correlate build timestamps. It's not that big a deal.

Reply:
Created an issue to track that: #4749